internal/service/alarms/internal/alarms_server.go

```diff
@@ -91,11 +92,43 @@ func (a *AlarmsServer) GetProbableCause(ctx context.Context, request api.GetProb
 	return nil, fmt.Errorf("not implemented")
 }
 
+// AmNotification Used internally, alertmanager from ACM calls here.
+// Note: the errors returned can also be view under alertmanager pod logs but also logging here for convenience
 func (a *AlarmsServer) AmNotification(ctx context.Context, request api.AmNotificationRequestObject) (api.AmNotificationResponseObject, error) {
-	// TODO: Implement the logic to handle the AM notification
-	slog.Debug("Received AM notification", "groupLabels", request.Body.GroupLabels)
-	for _, alert := range request.Body.Alerts {
-		slog.Debug("Alert", "fingerprint", alert.Fingerprint, "startsAt", alert.StartsAt, "status", alert.Status)
+	// Audit the table with full list of alerts in the current payload. If missing set them to resolve
+	if err := a.AlarmsRepository.ResolveNotificationIfNotInCurrent(ctx, request.Body); err != nil {
+		msg := "failed to resolve notification that are not present"
+		slog.Error(msg, "err", err)
+		return nil, fmt.Errorf("%s: %w", msg, err)
+	}
+
+	// Get the definition data based on current set of Alert names and managed cluster ID
+	alarmDefinitions, err := a.AlarmsRepository.GetAlarmDefinitions(ctx, request.Body)
+	if err != nil {
+		msg := "failed to get AlarmDefinitions"
+		slog.Error(msg, "err", err)
+		return nil, fmt.Errorf("%s: %w", msg, err)
+	}
+
+	// Combine possible definitions with events
+	aerModels := alertmanager.ConvertAmToAlarmEventRecordModels(request.Body, alarmDefinitions)
```

Comment:
I wonder if we should first build the aerModels, withouth the alarm definitions, then pass it to the GetAlarmDefinitions because we use it to get related alarm definitions and then add the for you have in ConvertAmToAlarmEventRecordModels to map the alerts with the definitions

Reply:
Yeah either way should work. Found it simpler to get all the definitions from the current alertmanager payload instead from aer. 

The idea is that aer is basically an combination of alertmanager playload + definition (which also happens to be the two params for `ConvertAmToAlarmEventRecordModels`.